src/logging/logging.h

```diff
@@ -88,38 +88,44 @@ constexpr const char* file_name(const char* const str) { return str_slant(str) ?
 #define LINEOUTPUTFORMAT "[{}:{}:{}] "
 #define LINEOUTPUTARGS file_name(__FILE__), __LINE__, __FUNCTION__
 
-#define LOGTRACEMOD_USING_LOGGER(mod, logger, msg, ...)                                                                \
-    if (auto& _l{logger}; _l && LEVELCHECK(mod, spdlog::level::level_enum::trace)) {                                   \
-        _l->trace(LINEOUTPUTFORMAT msg, LINEOUTPUTARGS, ##__VA_ARGS__);                                                \
-    }
+#define LOGTRACEMOD_USING_LOGGER(mod, logger, msg, ...) do {                                                           \
+        if (auto& _l{logger}; _l && LEVELCHECK(mod, spdlog::level::level_enum::trace)) {                               \
+            _l->trace(LINEOUTPUTFORMAT msg, LINEOUTPUTARGS, ##__VA_ARGS__);                                            \
+        }                                                                                                              \
+    } while(0)
```

Comment:
We have removed the trailing `;` here. Is there a reason for it? I believe consumers of this macro would be all of the SDS repos and not sure if all of them have a `;` at the end and so it could break those builds.

Reply:
I think there is a code sniffer in sonar that dislikes the extraneous `;`? We should just disable this check in sonar (which we can do), I agree that causing us to sanitize all the SDS code is not reasonable for something we know works and is not technically "wrong".

Reply:
This sounds good to me. Lets disable this check in sonar